src/controllers/keyboard/keyboardeventfilter.h

```diff
@@ -39,13 +42,13 @@ class KeyboardEventFilter : public QObject {
     };
 
     // Returns a valid QString with modifier keys from a QKeyEvent
-    QKeySequence getKeySeq(QKeyEvent *e);
+    QString getKeySeq(QKeyEvent* e);
+
     // List containing keys which is currently pressed
     QList<KeyDownInformation> m_qActiveKeyList;
-    // Pointer to keyboard config object
-    ConfigObject<ConfigValueKbd> *m_pKbdConfigObject;
-    // Multi-hash of key sequence to
-    QMultiHash<ConfigValueKbd, ConfigKey> m_keySequenceToControlHash;
+
+    // Clone of keyboard controller preset, containing keyboard mapping info
+    QSharedPointer<KeyboardControllerPreset> m_kbdPreset;
```

Comment:
Really a cloned instance? You only assign the shared pointer in slotSetKeyboardMapping, i.e. you are simply referencing a shared object.